src/export/csv/index.ts

```diff
@@ -0,0 +1,101 @@
+import { GraphQLClient } from "graphql-request";
+import omit from "lodash.omit";
+import { getFlowName, getLatestFlowGraph } from "../../flow";
+import { getSessionById } from "../../session";
+import { getBOPSParams } from "../bops/index";
+import { QuestionAndResponses, Response } from "../bops/model";
+import { CSVData } from "./model";
+
+export async function generateCSVData(
+  client: GraphQLClient,
+  sessionId: string
+): Promise<CSVData> {
+  const session = await getSessionById(client, sessionId);
+  const flow = await getLatestFlowGraph(client, session.flowId);
+  if (!flow)
+    throw new Error(
+      `Cannot get flow ${session.flowId}, therefore cannot generate CSV data.`
+    );
+  const flowName = await getFlowName(client, session.flowId);
+
+  const bopsData = getBOPSParams({
+    breadcrumbs: session.data.breadcrumbs,
+    flow: flow,
+    passport: session.data.passport,
+    sessionId: session.id,
+    flowName: flowName,
+  });
+
+  // format dedicated BOPS properties (eg `applicant_email`) as list of questions & responses to match proposal_details
+  //   omitting debug data and payload keys already in confirmation details
+  const summary: Record<string, any> = {
+    ...omit(bopsData, ["planx_debug_data", "files", "proposal_details"]),
+  };
+  const formattedSummary: { question: string; responses: Array<Response> }[] =
+    [];
+  Object.keys(summary).forEach((key) => {
+    formattedSummary.push({
+      question: key,
+      responses: [{ value: summary[key] }],
+    });
+  });
+
+  // similarly format file uploads as list of questions, responses, metadata
+  const formattedFiles: QuestionAndResponses[] = [];
+  bopsData["files"]?.forEach((file) => {
+    formattedFiles.push({
+      question: file.tags
+        ? `File upload: ${file.tags.join(", ")}`
+        : "File upload",
+      responses: [
+        { value: file.filename.split("/").pop() || "Unknown filename" },
+      ],
+      metadata: { notes: file.applicant_description || "" },
+    });
+  });
+
+  // gather key reference fields, these will be first rows of CSV
+  const references: { question: string; responses: Array<Response> }[] = [
+    {
+      question: "Planning Application Reference", // match language used on Confirmation page
+      responses: [{ value: sessionId }],
```

Comment:
I updated these references from the original implementation to adhere to the `responses` type, but realized that wrapping in `value` does break the HTML overview parsing in a couple places which I haven't updated yet in planx-document-templates.

To sense check: does having a consistent `responses` type always wrapped in value seem right here, or is `responses: string | Array<Response>` not a problem?